be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1903,13 +2015,17 @@ Status SegmentIterator::_next_batch_internal(vectorized::Block* block) {
             _block_rowids.resize(_opts.block_row_max);
         }
         _current_return_columns.resize(_schema->columns().size());
+        _converted_column_ids.resize(_schema->columns().size(), 0);
         for (size_t i = 0; i < _schema->num_column_ids(); i++) {
             auto cid = _schema->column_id(i);
             auto column_desc = _schema->column(cid);
             if (_is_pred_column[cid]) {
-                RETURN_IF_CATCH_EXCEPTION(_current_return_columns[cid] =
-                                                  Schema::get_predicate_column_ptr(
-                                                          *column_desc, _opts.io_ctx.reader_type));
+                auto storage_column_type = _storage_name_and_type[cid].second;
+                RETURN_IF_CATCH_EXCEPTION(
+                        _current_return_columns[cid] = Schema::get_predicate_column_ptr(
+                                _is_char_type[cid] ? FieldType::OLAP_FIELD_TYPE_CHAR
```

Comment:
Char type is special , since char type's datatype is same with string, both are `DataTypeString`, but `DataTypeString` only return `FieldType::OLAP_FIELD_TYPE_STRING`.

Reply:
comment add to code